src/wireviz/DataClasses.py

```diff
@@ -154,52 +174,74 @@ def __post_init__(self) -> None:
         self.ports_right = False
         self.visible_pins = {}
 
-        if self.style == 'simple':
+        if self.style == "simple":
             if self.pincount and self.pincount > 1:
-                raise Exception('Connectors with style set to simple may only have one pin')
+                raise Exception(
+                    "Connectors with style set to simple may only have one pin"
+                )
             self.pincount = 1
 
         if not self.pincount:
-            self.pincount = max(len(self.pins), len(self.pinlabels), len(self.pincolors))
+            self.pincount = max(
+                len(self.pins), len(self.pinlabels), len(self.pincolors)
+            )
             if not self.pincount:
-                raise Exception('You need to specify at least one, pincount, pins, pinlabels, or pincolors')
+                raise Exception(
+                    "You need to specify at least one, pincount, pins, pinlabels, or pincolors"
+                )
 
         # create default list for pins (sequential) if not specified
         if not self.pins:
             self.pins = list(range(1, self.pincount + 1))
 
         if len(self.pins) != len(set(self.pins)):
-            raise Exception('Pins are not unique')
+            raise Exception("Pins are not unique")
 
         if self.show_name is None:
-            self.show_name = not self.autogenerate # hide auto-generated designators by default
+            # hide designators for simple and for auto-generated connectors by default
+            self.show_name = self.style != "simple" and self.name[0:2] != "__"
 
         if self.show_pincount is None:
-            self.show_pincount = self.style != 'simple' # hide pincount for simple (1 pin) connectors by default
+            # hide pincount for simple (1 pin) connectors by default
+            self.show_pincount = self.style != "simple"
 
         for loop in self.loops:
-            # TODO: check that pins to connect actually exist
             # TODO: allow using pin labels in addition to pin numbers, just like when defining regular connections
             # TODO: include properties of wire used to create the loop
             if len(loop) != 2:
-                raise Exception('Loops must be between exactly two pins!')
+                raise Exception("Loops must be between exactly two pins!")
+            for pin in loop:
+                if pin not in self.pins:
```

Comment:
PR #288 claims to include such functionality, but it seems a bit unfinished, and I haven't checked how easy it might be to cherry-pick only that functionality. Unless you know about a quick-fix with existing code, then I guess we should postpone this functionality.

Reply:
Let's postpone please.